Update the automated release process description (#645)
submariner-io/submariner-website: src/content/development/release-process/_index.en.md

### Creating Releases

The following sections are an ordered series of steps to create a Submariner release.

The release process is mostly automated and uses a YAML file created in the
[releases repository](https://github.com/submariner-io/releases) that describes the release.
This file is updated for each step in the release process.

Once the changes for a step are reviewed and merged, a CI job will run to create the release(s) for the step and create the
required pull requests in preparation for the next step to be reviewed and merged. Once all these pull requests have been
merged, you can continue onto the next step.

> For most projects, after a release is created, another job will be initiated to build release artifacts and publish it to Quay.

Reviewer: ```suggestion
For most projects, after a release is created, another job will be initiated to build release artifacts and publish them to Quay.
```

I also think it's okay as-is, "publish to Quay".

Author: Reverting to what it was originally.

This will take several minutes. You can monitor the progress from the project's main page. In the branches/tags pull-down
above the file list heading, select the tag for the new version. A small yellow circle icon should be present to the right

> of the file list heading which indicates a job is in progress. You can click on it to see details. There may be several checks

Reviewer: The guidance I'm seeing is that it should remain as-is.

https://helpcenter.veeam.com/docs/styleguide/tw/click.html

{{% /notice %}}

The command runs, gathers the data for the release, updates the release YAML and pushes it for review. Once the review process is done,

> and the automated release steps on the CI have finished successfully, merge the PR. Pull requests will then be created for all dependent

Reviewer: ```suggestion
, merge the PR. Pull requests will then be created for all dependent
```

The original meant the automation that runs after the PR is merged, your change makes it about the tests on the PR. I think the result is a bit redundant, we don't need to explain that tests have to pass as a part of the review process and you talk about the post-merge automation later.

> projects to update them to the new version pertaining to the current step, if necessary, in preparation for the next step. The automation

Reviewer: ```suggestion
projects to update them to the new version. The automation
```

> and the automated release steps on the CI have finished successfully, the PR is merged. Pull requests will be created for all dependent
> projects to update them to the new version in preparation for the subsequent  steps. The automation will leave a comment with a list of
> them. Make sure all those PRs are merged and their release jobs pass. Then proceed with the next step, by simply **running the same

Reviewer 2: ```suggestion
and the automated release steps on the CI have finished successfully, merge the PR. Pull requests will then be created for all dependent
projects to update them to the new version pertaining to the current step, if necessary, in preparation for the next step. The automation will also leave a comment with a list of
the dependent project PRs in the release PR . Make sure all those PRs are merged and their release jobs pass then proceed to the next step, by simply **running the same
```

Reviewer: > pertaining to the current step, if necessary, in preparation for the next step

Sorry to have conflicting advice for this bit @Jaanki, but I really think it's unnecessary, that it adds cognitive load without adding anything useful.

> will also leave a comment with a list of the dependent project PRs in the release PR . Make sure all those PRs are merged and their

Reviewer: ```suggestion
will leave a comment with a list of the version-bump PRs for dependent project the release PR that was just merged. Make sure all those PRs are merged and their
```

> release jobs pass then proceed to the next step, by simply **running the same command again to advance to the next stage**.

Reviewer: ```suggestion
release jobs pass (see the Actions tab of the repository on GitHub) then **proceed to the next release phase by running the same command again**.
```
